File: nomad-samples/admin/py/content_api/contents/get_content.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests

def get_content(AUTH_TOKEN, ID, CONTENT_DEFINITION_ID, IS_REVISION):

    API_URL = f"{ADMIN_API_URL}/content/{ID}?contentDefinitionId={CONTENT_DEFINITION_ID}&isRevision={IS_REVISION}"

    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        RESPONSE = requests.get(API_URL, headers=HEADERS)

        if (not RESPONSE.ok):
            raise Exception()
        
        return RESPONSE.json()
    
    except:
        api_exception_handler(RESPONSE, f"Getting content {ID} failed")

File: nomad-samples/admin/js/contents-api/contents/get-content.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function getContent(AUTH_TOKEN, CONTENT_DEFINITION_ID, ID, IS_REVISION) {
    
    let API_URL = `${prjConstants.ADMIN_API_URL}/content/${ID}?contentDefinitionId=${CONTENT_DEFINITION_ID}&isRevision=${IS_REVISION}`

    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
    HEADERS.append("Content-Type", "application/json");

    // Send the request
    const RESPONSE = await fetch(API_URL, {
        method: "GET",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        return await RESPONSE.json();
    }

    await apiExceptionHandler(RESPONSE, `Getting Content ${ID} failed`);
}